core/domain/classifier_domain.py

```diff
@@ -504,7 +470,126 @@ def validate(self):
                 'Expected state_name to be a string, received %s' % (
                     self.state_name))
 
-        if not isinstance(self.job_id, python_utils.BASESTRING):
+        self._algorithm_id_to_job_id_map.validate()
+
+
+class AlgorithmIDToJobIDMapping(python_utils.OBJECT):
+    """Domain object for a algorithm_id to training job_id map.
+
+    Each algorithm_id is mapped to a unique job_id which stores the metadata
+    related to the training job for that algorithm_id.
+
+    Attributes:
+        algorithm_id_to_job_id_map. dict. Mapping of algorithm IDs to
+            corresponding unique trainnig job IDs.
+    """
+
+    def __init__(self, algorithm_id_to_job_id_map):
+        """Constructs a AlgorithmIDToJobIDMapping domain object.
+
+        Args:
+            algorithm_id_to_job_id_map: dict. Mapping of algorithm IDs to
```

Comment:
We typically don't name stuff as *_map; consider algorithm_ids_to_job_ids

Reply:
Done